### PR TITLE
Issue #89/#153: Update find/sort command for Seniority, Application Status and Interview Status

### LINKED ISSUE #89: Update `sort` command for `Candidate`'s Seniority, Application Status and Interview Status
To add sorting functionality across the Application Status and Interview Status fields as well.

### LINKED ISSUE #153: Update `find` command for `Candidate`'s Seniority, Application Status and Interview Status

### PR DESCRIPTION
**Pull request by charmainehly**
Currently, the sort and find command are only able to function with previous candidate fields.

The user is therefore not able to search across seniority, interview status and application status.

In this PR: Let's add the switch-case statements for the fields above, and also the relevant ContainsKeywordsPredicate classes for the find command.

Note: This commit also fixes the implementation for updateSortedCandidateList() in the ModelManager. Error in the previous implementation results in the original addressbook candidate list being modified.

Closes #89 and Closes #153

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/FindCommandParser.java [hunk 1]
```diff
@@ -49,17 +52,23 @@ public FindCommand parse(String args) throws ParseException {
...
             return new FindCommand(new NameContainsKeywordsPredicate(keywords));
-        case "candidate":
-        case "":
-            return new FindCommand(new CandidateContainsKeywordsPredicate(keywords));
         case "phone":
             return new FindCommand(new PhoneContainsKeywordsPredicate(keywords));
+        case "seniority":
+            return new FindCommand(new SeniorityContainsKeywordsPredicate(keywords));
         case "studentid":
             return new FindCommand(new StudentIdContainsKeywordsPredicate(keywords));
```
Comment (domlimm) [inline src/main/java/seedu/address/logic/parser/FindCommandParser.java:73]: As always, neat!

File: src/main/java/seedu/address/logic/parser/SortCommandParser.java [hunk 2]
```diff
@@ -34,18 +34,27 @@ public SortCommand parse(String args) throws ParseException {
         Comparator<Candidate> sortComparator;
 
         switch (sortKey) {
+        case "applicationstatus":
+            sortComparator = Comparator.comparing(l -> l.getApplicationStatus().toString().toLowerCase());
+            break;
```
Comment (domlimm) [inline src/main/java/seedu/address/logic/parser/SortCommandParser.java:39]: Nice lambda! 🐐

File: src/test/java/seedu/address/logic/parser/FindCommandParserTest.java [hunk 3]
```diff
@@ -80,6 +83,28 @@ public void parse_validArgs_returnsFindStudentIdCommand() {
         assertParseSuccess(parser, "    k/E0324 \t \t  k/149  \t  f/studentid \t", expectedFindStudentIdCommand);
     }
 
+    @Test
+    public void parse_validArgs_returnsFindStatusCommand() {
```
Comment (domlimm) [inline src/test/java/seedu/address/logic/parser/FindCommandParserTest.java:87]: Really creative test cases to sieve out edge cases! 💯

**Review by domlimm — APPROVED** [review]
Nice job @charmainehly! Thank you for the work done! Nothing violating at all, all clean and good. Approved! 👍🏻

**Review by lzan98 — APPROVED** [review]
LGTM! Sorry I think I missed this out amongst the PRs 😅
